WordPress/src/main/java/org/wordpress/android/ui/mediapicker/MediaPickerViewModel.kt

```diff
@@ -431,11 +460,13 @@ class MediaPickerViewModel @Inject constructor(
         object Hidden : ActionModeUiModel()
     }
 
-    sealed class SearchUiModel {
-        object Collapsed : SearchUiModel()
-        data class Expanded(val filter: String) : SearchUiModel()
+    sealed class SearchUiModel(val isVisible: Boolean) {
+        class Collapsed(isVisible: Boolean) : SearchUiModel(isVisible)
+        class Expanded(isVisible: Boolean, val filter: String) : SearchUiModel(isVisible)
```

Comment:
I think I'd probably introduce third child state here: Collapsed, Expanded and Hidden? Does it make sense to keep the information about search when it's not visible? I'd also keep data classes just to be sure (for the equality check in the `distinct`)

Reply:
Good point (especially the one related to data class and `distinct` usage 👍)...changed here 931218e1dd536536d50e0c1224cd1340b6213ac4